### PR TITLE
Add lancedb-haystack to integrations.

### PR DESCRIPTION
**Pull request by alanmeeson**
This PR adds the LanceDB Haystack Document Store from [https://github.com/alanmeeson/lancedb-haystack](https://github.com/alanmeeson/lancedb-haystack). 

Notes from Discord discussion:
There are some interesting quirks due to the nature of LanceDB; you need to specify a schema for your metadata so that it can be turned into a pyarrow struct (and thus be searchable).  Also the precision of the embedding vectors seems to be limited to an error of around 10^-7 compared to it's python form - feels like some sort of difference between the float representations. 

Please take a look when you have time, and let me know if I should change anything.

Thanks!

### CODE REVIEW

File: integrations/lancedb.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+- [License](#license)
+
+## Overview
+LanceDB-Haystack is an embedded [LanceDB](https://lancedb.github.io/lancedb/) backed Document Store for [Haystack 2.X](https://github.com/deepset-ai/haystack/).
+
+## Installation
+
+The current simplest way to get LanceDB-Haystack is to install from GitHub via pip:
+
+```pip install lancedb-haystack```
```
Comment (bilgeyucel) [inline integrations/lancedb.md:29]:
```suggestion
```shell
pip install lancedb-haystack

```

**Review by bilgeyucel — COMMENTED** [review]
Thanks for the PR @alanmeeson. I left one minor comment. Also feel free to add a logo that will be displayed on a dark background 😊

File: integrations/lancedb.md (hunk 1 above)
Comment (bilgeyucel) [reply integrations/lancedb.md:29]: It's quite hard to show it with github's markdown syntax. This is what I mean
image

Comment (alanmeeson) [reply integrations/lancedb.md]: Updated :heavy_check_mark:

**Comment by alanmeeson** [comment]
I've updated the pip install block to be tagged as shell,  got in touch with the LanceDB team and sourced the official 'light on dark' logo, and added a link to the new API docs.

**Review by bilgeyucel — APPROVED** [review]
🚀

**Pull request merged**